### PR TITLE
Support disabling specular highlights from directional lights

### LINKED ISSUE #5157: Diffuse Only Light feature
Provide as much information as possible. Include (where applicable):

[https://playcanvas.com/editor/scene/1692776](url)

image

The light reflection looks okay in some case, but it's not fine when it comes to high quality product rendering.

It will be great if we can use Diffuse Light Only without giving up specular.

Hoping this feature is available soon.

### PR DESCRIPTION
**Pull request by ellthompson**
This PR adds the following option to the light component:

```javascript
class LightComponent {
/**
 * @property {boolean} affectSpecularity If enabled and the light type is pc.LIGHTTYPE_DIRECTIONAL, material specularity
 * will not be affected by this light. Defaults to true.
 */
}
```

This property can be continuously updated and affects only the light source it is associated with.

Fixes #5157

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by Maksims** [comment]
Is it common to use `disableThing` semantic for material properties?
Other properties commonly used: `useThing`.
Or just `thing`.
Also `isThing` used in few other places.

`isSpecular`, `useSpecularity` or something?
Unfortunately some non-native speakers night find "highlight" word unrelated to "specularity".

File: src/scene/shader-lib/programs/lit-shader.js [hunk 1]
```diff
@@ -1303,10 +1305,10 @@ class LitShader {
 
                     // area light
                     if (options.useClearCoat) {
-                        backend.append(`    ccSpecularLight += ccLTCSpecFres * get${shapeString}LightSpecular(litShaderArgs.clearcoat.worldNormal, dViewDirW) * dAtten * light${i}_color` + (usesCookieNow ? " * dAtten3" : "") + ";");
+                        backend.append(`    ccSpecularLight += ccLTCSpecFres * get${shapeString}LightSpecular(litShaderArgs.clearcoat.worldNormal, dViewDirW) * dAtten * light${i}_color * light${i}_specularColorFactor` + (usesCookieNow ? " * dAtten3" : "") + ";");
```
Comment (GSterbrant) [inline src/scene/shader-lib/programs/lit-shader.js:1308]: Like Martin said, I would probably just change this chunk to do something like `... light${i}_color * ${specularSwitch}...` and then just set specularSwitch to 1 normally or 0 if we disable specular highlights.

**Comment by ellthompson** [comment]
> Is it common to use `disableThing` semantic for material properties? Other properties commonly used: `useThing`. Or just `thing`. Also `isThing` used in few other places.
> 
> `isSpecular`, `useSpecularity` or something? Unfortunately some non-native speakers night find "highlight" word unrelated to "specularity".

I've updated this to `affectSpecularity` which is a style already used in the light component.

File: src/framework/components/light/component.js [hunk 2]
```diff
@@ -375,6 +377,9 @@ function _defineProps() {
     _defineProperty('shape', LIGHTSHAPE_PUNCTUAL, function (newValue, oldValue) {
         this.light.shape = newValue;
     });
+    _defineProperty('affectSpecularity', true, function (newValue, oldValue) {
```
Comment (mvaligursky) [inline src/framework/components/light/component.js:380]: could you please add this new property to lightComponentProps in types-fixup as well

**Review by mvaligursky — APPROVED** [review]
I left a small comment, otherwise looks great.
@GSterbrant - can you please double check the shader changes.

File: src/framework/components/light/component.js (hunk 2 above)
Comment (ellthompson) [reply src/framework/components/light/component.js:380]: Done 👍

**Comment by GSterbrant** [comment]
Looks good! Let's merge.

**Pull request merged**